internal/internal_workflow_testsuite_test.go

```diff
@@ -4031,3 +4031,49 @@ func (s *WorkflowTestSuiteUnitTest) Test_WorkflowGetCurrentHistoryLength() {
 	s.NoError(env.GetWorkflowResult(&result))
 	s.Equal(17, result)
 }
+
+func (s *WorkflowTestSuiteUnitTest) Test_ActivityWithStartToCloseTimeout() {
+	timeout := 100 * time.Millisecond
+
+	timeoutActivity := func(ctx context.Context) error {
+		time.Sleep(timeout * 2)
+		return nil
+	}
+
+	noTimeoutActivity := func(ctx context.Context) error {
```

Comment:
The low timing resolution/granularity of this test will lead to flakiness on an overloaded worker. I think you can just remove the no-timeout assertion part.